src/analysis/inner_product_space/l2_space.lean

```diff
@@ -242,69 +249,115 @@ begin
     exact hV.linear_isometry.isometry.uniform_inducing.is_complete_range.is_closed }
 end
 
-/-- A mutually orthogonal family of complete subspaces of `E`, whose range is dense in `E`, induces
-a isometric isomorphism from E to `lp 2` of the subspaces.
+end orthogonal_family
+
+section is_hilbert_sum
+
+variables (𝕜 E) (V : Π i, G i →ₗᵢ[𝕜] E) (F : ι → submodule 𝕜 E)
+include cplt
+
+/-- Given a family of Hilbert spaces `G : ι → Type*`, a Hilbert sum of `G` consists of a Hilbert
+space `E` and an orthogonal family `V : Π i, G i →ₗᵢ[𝕜] E` such that the induced isometry
+`Φ : lp G 2 → E` is surjective.
+
+Keeping in mind that `lp G 2` is "the" external Hilbert sum of `G : ι → Type*`, this is analogous
+to `direct_sum.is_internal`, except that we don't express it in terms of actual submodules. -/
+@[protect_proj] structure is_hilbert_sum : Prop :=
+(orthogonal_family : orthogonal_family 𝕜 V)
+(surjective_isometry : function.surjective (orthogonal_family.linear_isometry))
+
+variables {𝕜 E V}
+
+/-- If `V : Π i, G i →ₗᵢ[𝕜] E` is an orthogonal family such that the supremum of the ranges of
+`V i` is dense, then `(E, V)` is a Hilbert sum of `G`. -/
+lemma orthogonal_family.is_hilbert_sum [Π i, complete_space $ G i]
```

Comment:
Maybe name `is_hilbert_sum.mk` to match the corresponding result for Hilbert bases?  Also, could you please use the named constructor `{ orthogonal_family := hVortho, surjective_isometry := _ }` rather than the anonymous constructor?